bbolt/bbolt.go

```diff
@@ -104,9 +104,13 @@ type store struct {
 }
 
 func (b *store) Close(ctx context.Context) error {
-	return util.CallWithTimeout(ctx, b.db.Close, func() {
+	err := util.CallWithTimeout(ctx, b.db.Close, func() {
 		b.log.Error("Closing of BBolt store timed out, store may not shut down correctly.")
 	})
+	if err != nil {
```

Comment:
Would be simpler to have `DatabaseError` return `nil` when the given error is `nil`. Would remove many of the `if` checks on `err`.

Maybe not safer, but syntactic sugar to improve readability.

Reply:
I've looked at this, but to me it wasn't necessarily more readable. 
Left it out for now since adding this later is going to be a lot easier than removing it.